### ISSUE TITLE
Can't use keyword AND in custom sql in moor

### ISSUE THREAD

**Issue by jims57** [simolus3/moor#891]
When using multi where conditions, I need to use `and` keyword, but it shows error when building by command:
`flutter packages pub run build_runner build --delete-conflicting-outputs`

<img width="1379" alt="image" src="https://user-images.githubusercontent.com/1920873/96861592-e126d300-1496-11eb-8e30-0510a6ea0cf2.png">

BTW, when I exchange the positions from this:

```dart
SELECT * FROM notes n WHERE CASE WHEN 1 = :isReviewFolderSelected THEN n.nextReviewTime IS NOT NULL ELSE n.nextReviewTime IS NULL END and n.folderId = :selectedFolderId;
```

to this:
```dart
SELECT * FROM notes n WHERE n.folderId = :selectedFolderId and CASE WHEN 1 = :isReviewFolderSelected THEN n.nextReviewTime IS NOT NULL ELSE n.nextReviewTime IS NULL END;
```

The error become about `CASE` keyword as follows:
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/1920873/96862011-82158e00-1497-11eb-9c26-3081b6696c82.png">

**Comment by simolus3** [comment]
It looks like `CASE` has a precedence that's way too low in the `sqlparser` package.

Until I fix this, you could put the `CASE` expression in parenthesis (e.g. `and (CASE WHEN 1 ... END)`)

**Comment by simolus3** [comment]
I found and fixed the problem, thanks for the report! This will be fixed in the next moor update (hopefully coming soon).

**Comment by jims57** [comment]
@simolus3 Does `moor: ^3.4.0` fix this bug mentioned at this post?

**Comment by simolus3** [comment]
It's fixed in `sqlparser: 0.11.0`, which you'll get with `moor_generator: ^3.4.0`, yes.
